examples/address_generator/src/main.rs

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::{App, Arg};
+
+use transact::contract::address::{
+    double_key_hash::DoubleKeyHashAddresser, key_hash::KeyHashAddresser,
+    triple_key_hash::TripleKeyHashAddresser, Addresser,
+};
+
+fn main() -> Result<(), String> {
+    let matches = App::new(clap::crate_name!())
+        .version(clap::crate_version!())
+        .author(clap::crate_authors!())
+        .about(clap::crate_description!())
+        .arg(
+            Arg::with_name("prefix")
+                .long("prefix")
+                .takes_value(true)
+                .required(true)
+                .help("the prefix of the radix address"),
+        )
+        .arg(
+            Arg::with_name("keys")
+                .required(true)
+                .takes_value(true)
+                .max_values(3)
+                .help(
+                    "the natural keys used to compute the radix address, \
+                     must supply 1 - 3 separate keys",
+                ),
+        )
+        .get_matches();
+
+    // The `prefix` will be used to construct the Addresser
+    let prefix = matches
+        .value_of("prefix")
+        .expect("Unable to get `prefix` arg")
+        .to_string();
+
+    let keys: Vec<_> = matches
+        .values_of("keys")
+        .expect("Unable to get `keys` arg")
+        .collect();
+
+    // The Addresser implementation to be constructed depends on the amount of `keys` supplied.
+    // Once the appropriate Addresser implementation is constructed, the radix address will be
+    // calculated and printed out.
+    match keys.len() {
+        1 => {
+            let addresser = KeyHashAddresser::new(prefix);
+            let addr = addresser
+                .compute(&keys[0].to_string())
+                .map_err(|e| format!("Unable to compute address with KeyHashAddresser: {}", e))?;
+            println!("{}", addr);
+        }
+        2 => {
+            let addresser = DoubleKeyHashAddresser::new(prefix, None);
```

Comment:
Instead of None here, we should accept a command line argument to be able to set it.

Reply:
I addressed this by adding a `lengths` arg